cypress/integration/app_spec.ts

```diff
@@ -38,6 +50,35 @@ describe("Home Page", () => {
   });
 });
 
+describe("Pagination element functionality", () => {
+  beforeEach(() => {
+    setupServerMocks();
+  });
+
+  it("checks if next page button shows next page of results", () => {
+    cy.visit("/search?q=hello");
+    cy.get(`[data-e2e-id="SoundList-track-name"]`)
+      .first()
+      .should("be.visible")
+      .should("have.text", "hello");
+    cy.get(`[data-e2e-id="SoundList-track-url"]`)
+      .first()
+      .should("be.visible")
+      .should("have.attr", "href", "/sound/123");
+    cy.get(`[data-e2e-id="first-page-button"]`).should("be.visible");
+    cy.get(`[data-e2e-id="next-page-button"]`).should("be.visible").click();
+    cy.wait(2000); // This is necessary to give enough time to re-render.
+    cy.get(`[data-e2e-id="SoundList-track-name"]`)
+      .first()
+      .should("be.visible")
+      .should("have.text", "hello.wav");
+    cy.get(`[data-e2e-id="SoundList-track-url"]`)
+      .first()
+      .should("be.visible")
+      .should("have.attr", "href", "/sound/343161");
```

Comment:
After the second page of search results has enough time to render, we verify that the first track on that page has the appropriate `text` and `href`.